nautobot/ui/src/components/pagination.js

```diff
@@ -1,93 +1,159 @@
-                if (pageNumber === "...") {
-                    return (
-                        <li style={li_css}>
-                            <Button variant="secondary">&#8230;</Button>
-                        </li>
-                    );
-                }
-                // pageNumber (starts from 1) is always one greater than the currentPage (starts from 0)
-                // we add one here to make sure that the correct pageNumber is highlighted.
-                if (pageNumber === currentPage + 1) {
-                    return (
-                        <li
-                            style={li_css}
-                            onClick={() => onPageNumberChange(pageNumber)}
-                        >
-                            <Button variant="primary">{pageNumber}</Button>
-                        </li>
-                    );
-                } else {
-                    return (
-                        <li
-                            style={li_css}
-                            onClick={() => onPageNumberChange(pageNumber)}
-                        >
-                            <Button variant="secondary">{pageNumber}</Button>
-                        </li>
-                    );
-                }
-            })}
-            <li style={li_css} onClick={() => onPageNumberChange(lastPage)}>
-                <Button variant="secondary">{">"}</Button>
-            </li>
-        </ul>
+        <Flex align="center">
+            <Box
+                color="gray-3"
+                pl="xs"
+                pr="xs"
+                mr="lg"
+                fontSize="md"
+                border="1px"
+                borderColor="gray-1"
+                borderRadius="sm"
+            >
+                {totalDataCount} rows
+            </Box>
+            <Text color="gray-3" whiteSpace="nowrap">
+                You are on page
+            </Text>
+            {trueCurrentPage > firstPage ? (
+                <IconButton
+                    onClick={() => onPageNumberChange("<")}
+                    icon={<ChevronLeftIcon />}
+                    color="gray-4"
+                    variant="ghost"
+                    pl="xs"
+                    _hover={{
+                        transform: "scale(1.2)",
+                        color: "#007DFF",
+                    }}
+                />
+            ) : (
+                <Box pl="xs" />
+            )}
+            <FormControl
+                isInvalid={isInputInvalid}
+                align="center"
+                pl="xs"
+                pr="xs"
+            >
+                <Input
+                    type="number"
+                    placeholder={pageNumber}
+                    textAlign="center"
+                    width="50px"
+                    onBlur={handleOnBlur}
+                />
+                {isInputInvalid && (
+                    <FormErrorMessage>Page out of range</FormErrorMessage>
+                )}
```

Comment:
When this is True, the page does not appear to be V center aligned; perhaps adjusting this to absolute will help.:
![Screenshot 2023-06-30 at 10 07 12 AM](https://github.com/nautobot/nautobot/assets/94907097/d1c8fae8-6da3-4d9d-893c-4edc29ca3d00)